cocoa-mhlw/cocoa
# [WIP]Fix issue that background task fail on Android.

**Work In Progress, DO NOT MERGE**

## Purpose
Fix issue #25

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
 1.  Launch the app
 2. Reboot device
 3. Watch Logcat
 4. Wait 16 minutes
 5. Confirm background worker succeed on Logcat. (e.g. [common_service_locator.log](https://github.com/cocoa-mhlw/cocoa/files/6056573/common_service_locator.log))

----

#25 において、画面を介さずに起動したバックグラウンドタスクでは、`App`に依存した`DependencyService.Resolve`は使用できないことがわかった。

このPullRequestでは、`DependencyService.Resolve`に代わり`DryIoc.CommonServiceLocator`で依存性の注入（正確にはInversion of Control）を行う経路を新たに**追加した**。

また、`ExposureNotificationHandler`の経路上で`App`に依存したクラスや処理についても適宜変更を加えている。

## 今後の課題（iOS対応・テスト）
現状、`DryIoc.CommonServiceLocator`の初期化は`Covid19Radar.Droid.MainApplicaiton`のみで行っている。iOSで同等の処理を追加する必要がある。
